autonomic-plugin-common/src/main/java/br/com/autonomiccs/autonomic/plugin/common/services/AutonomicClusterManagementHeuristicService.java

```diff
@@ -35,19 +35,24 @@
 import br.com.autonomiccs.autonomic.administration.algorithms.ClusterAdministrationHeuristicAlgorithm;
 import br.com.autonomiccs.autonomic.administration.algorithms.impl.ClusterManagementDummyAlgorithm;
 
+/**
+ * This class manages the instantiation of cluster administration heuristics, classes that implement
+ * the interface {@link ClusterAdministrationHeuristicAlgorithm}.
+ */
 @Service("autonomicClusterManagementHeuristicService")
 public class AutonomicClusterManagementHeuristicService {
 
     public final static String CLUSTER_ADMINISTRATION_ALGORITHMS_IN_CONFIGURATION_KEY = "autonomiccs.clustermanager.algorithm";
 
     private final Logger logger = LoggerFactory.getLogger(getClass());
-    private final Map<String, Class<? extends ClusterAdministrationHeuristicAlgorithm>> algorithmsMap = new HashMap<>();
+    protected final Map<String, Class<? extends ClusterAdministrationHeuristicAlgorithm>> algorithmsMap = new HashMap<>();
 
     @Autowired
-    private ConfigurationDao configurationDao;
+    protected ConfigurationDao configurationDao;
 
     /**
-     * TODO
+     * Returns the instance of the Administration algorithm configured by the user; if could not
```

Comment:
Returns the instance of the Administration algorithm configured by the system administrator; if it can not find the configured heuristic it returns the {@link ClusterManagementDummyAlgorithm}